src/languages/gml.js

```diff
@@ -2994,135 +898,569 @@ export default function(hljs) {
-    "view_camera",
-    "view_current",
-    "view_enabled",
-    "view_hport",
-    "view_surface_id",
-    "view_visible",
-    "view_wport",
-    "view_xport",
-    "view_yport",
-    "visible",
-    "vspeed",
-    "webgl_enabled",
-    "working_directory",
-    "x",
-    "xprevious",
-    "xstart",
-    "y",
-    "yprevious",
-    "ystart"
+    "rollback_confirmed_frame",
+    "rollback_api_server",
+    "player_avatar_sprite",
+    "player_avatar_url",
+    "player_user_id",
+    "player_type"
+  ];
+  /**
+   * Regex for some sort of identifier - i.e, a valid name of something in code.
+   */
+  const VALID_IDENTIFIER_RE = /[a-zA-Z_][a-zA-Z0-9_]*/;
+  /**
+   * Regex for a dot separating some LHS and RHS expression with optional whitespace (as this is
+   * supported in the engine.)
+   */
+  const DOT_ACCESSOR_RE = /\b\.\b/;
+
+  /**
+   * Expressions, which form part of a valid statement.
+   * @type {Object[]}
+   */
+  const EXPRESSION = [];
+
+  /**
+   * A single-line comment.
+   */
+  const COMMENT_LINE = hljs.COMMENT('//', /$/);
+
+  const JSDOC_TYPE_PARAM_RE = /{ *[a-zA-Z_][a-zA-Z0-9_\.<>\|]* *} */;
+
+  const JSDOC_ANNOTATIONS = [
+    {
+      match: /@((desc(ription)?)|ignore|pure|deprecated|(func(tion)?))\b/,
+      scope: "doctag"
+    },
+    {
+      match: [
+        /@(self|context) */,
+        JSDOC_TYPE_PARAM_RE
+      ],
+      scope: {
+        1: "doctag",
+        2: "type",
+      }
+    },
+    {
+      match: [
+        /(@return)s? */,
+        JSDOC_TYPE_PARAM_RE
+      ],
+      scope: {
+        1: "doctag",
+        2: "type",
+      },
+      relevance: 0,
+    },
+    {
+      match: [
+        /@((param(eter)?)|arg(ument)?) */,
+        JSDOC_TYPE_PARAM_RE,
+        /\[?/,
+        VALID_IDENTIFIER_RE,
+        /(( *=[^\n]+?)?\])? *?/
+      ],
+      scope: {
+        1: "doctag",
+        2: "type",
+        3: "params",
+        4: "params",
+        5: "params",
+      },
+      relevance: 0,
+    }
+  ];
+
+  /**
+   * A comment that documents a function using the same style as JSDoc for JavaScript.
+   */
+  const COMMENT_JSDOC = {
+    begin: /\/\/\//,
+    end: /$/,
+    scope: "comment",
+    relevance: 0,
+    contains: JSDOC_ANNOTATIONS,
+  };
+
+  /**
+   * A comment that documents a function using the same style as JSDoc for JavaScript.
+   */
+  const COMMENT_BLOCK_JSDOC = {
+    begin: /\/\*\*/,
+    end: /\*\//,
+    scope: "comment",
+    relevance: 0,
+    contains: JSDOC_ANNOTATIONS
+  };
+
+  /**
+   * Modes for the types of comments supported in GML.
+   */
+  const COMMENT = {
+    variants: [
+      COMMENT_JSDOC,
+      COMMENT_BLOCK_JSDOC,
+      COMMENT_LINE,
+      hljs.C_BLOCK_COMMENT_MODE,
+    ]
+  };
+
+  /**
+   * A template string substitution. `contains` is filled in after `EXPRESSION` is defined due to
+   * nesting.
+   */
+  const STRING_SUBSTITUTION = {
+    begin: /{/,
+    end: /}/,
+    beginScope: "char.escape",
+    endScope: "char.escape",
```

Comment:
I don't know that we have a scope for this other than `punctuation`, but pretty sure `char.escape` doesn't fit.  That's for things like `\n`, etc...